src/coreComponents/constitutive/fluid/multifluid/PVTDriverRunTest.hpp

```diff
@@ -65,18 +74,60 @@ void PVTDriver::runTest( FLUID_TYPE & fluid, arrayView2d< real64 > const & table
   integer const numSteps = m_numSteps;
   using ExecPolicy = typename FLUID_TYPE::exec_policy;
   forAll< ExecPolicy >( composition.size( 0 ),
-                        [numPhases, numSteps, kernelWrapper, table, composition] GEOS_HOST_DEVICE ( localIndex const i )
+                        [numPhases, numComponents, numSteps, kernelWrapper, table, composition,
+                         molarWeights=componentMolarWeights.toViewConst(),
+                         outputCompressibility=m_outputCompressibility,
+                         outputPhaseComposition=m_outputPhaseComposition]
+                        GEOS_HOST_DEVICE ( localIndex const i )
   {
+    constexpr real64 epsilon = LvArray::NumericLimits< real64 >::epsilon;
+
+    // Pressure derivative index
+    constexpr integer dPIndex = constitutive::multifluid::DerivativeOffset::dP;
+
+    // Index for start of phase properties
+    integer const PHASE = outputCompressibility != 0 ? TEMP + 3 : TEMP + 2;
+
     for( integer n = 0; n <= numSteps; ++n )
     {
       kernelWrapper.update( i, 0, table( n, PRES ), table( n, TEMP ), composition[i] );
-      table( n, TEMP + 1 ) = kernelWrapper.totalDensity()( i, 0 );
+      real64 const totalDensity = kernelWrapper.totalDensity()( i, 0 );
+      table( n, TEMP + 1 ) = totalDensity;
+
+      if( outputCompressibility != 0 )
+      {
+        real64 compressibility = kernelWrapper.dTotalDensity()( i, 0, dPIndex ) / totalDensity;
+        table( n, TEMP + 2 ) =  LvArray::math::max( 0.0, compressibility );
```

Comment:
Perhaps we should add a `kernelWrapper.compressibility( i, 0 )` function to the fluid model itself, rather than doing this computation here?